Introduce InternalWritableStreamWriter as a first step towards supporting piping from readable byte streams

youennf opened the pull request:
#### 4f7c00e3626b9ed202a12f86969befd8b1ae7261
<pre>
Introduce InternalWritableStreamWriter as a first step towards supporting piping from readable byte streams
<a href="https://rdar.apple.com/161244618">rdar://161244618</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=299437">https://bugs.webkit.org/show_bug.cgi?id=299437</a>

Reviewed by Chris Dumez.

* Source/WebCore/bindings/js/InternalWritableStreamWriter.cpp: Added.
(WebCore::invokeWritableStreamWriterFunction):
(WebCore::acquireWritableStreamDefaultWriter):
(WebCore::writableStreamDefaultWriterGetDesiredSize):
(WebCore::writableStreamDefaultWriterCloseWithErrorPropagation):
(WebCore::writableStreamDefaultWriterRelease):
(WebCore::writableStreamDefaultWriterWrite):
(WebCore::InternalWritableStreamWriter::onClosedPromiseRejection):
(WebCore::InternalWritableStreamWriter::onClosedPromiseResolution):
(WebCore::InternalWritableStreamWriter::whenReady):
* Source/WebCore/bindings/js/InternalWritableStreamWriter.h: Added.

Canonical link: <a href="https://commits.webkit.org/300679@main">https://commits.webkit.org/300679@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8d5a449c4dc142456264a57598e5689c7f92c646

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123494 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43209 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33905 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130222 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75634 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3760b569-2291-4f93-a76b-45f1e8e8d4ff) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125371 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43932 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51803 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93879 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62323 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e5117d4a-da71-4c6c-8eca-5e907f767087) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126447 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35010 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110487 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74511 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/25c88400-152d-42a9-9c1d-3939cdcce070) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33980 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28645 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73729 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104726 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28870 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132932 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50445 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38404 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102367 "Found 1 new test failure: imported/blink/fast/pagination/viewport-y-horizontal-bt-rtl.html (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50820 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106709 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102218 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25980 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47575 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25806 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47253 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50299 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56060 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49773 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53120 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51448 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/bindings/js/InternalWritableStreamWriter.h:
```diff
@@ -0,0 +1,58 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY CANON INC. ``AS IS'' AND ANY
```
Why do your copyright headers say CANON? Likely applies elsewhere too.

youennf replied:
Fixed, thanks for spotting it.

cdumez reviewed (approved):
LGTM with copyright fix.

youennf committed:
Introduce InternalWritableStreamWriter as a first step towards supporting piping from readable byte streams
rdar://161244618
https://bugs.webkit.org/show_bug.cgi?id=299437

Reviewed by Chris Dumez.

* Source/WebCore/bindings/js/InternalWritableStreamWriter.cpp: Added.
(WebCore::invokeWritableStreamWriterFunction):
(WebCore::acquireWritableStreamDefaultWriter):
(WebCore::writableStreamDefaultWriterGetDesiredSize):
(WebCore::writableStreamDefaultWriterCloseWithErrorPropagation):
(WebCore::writableStreamDefaultWriterRelease):
(WebCore::writableStreamDefaultWriterWrite):
(WebCore::InternalWritableStreamWriter::onClosedPromiseRejection):
(WebCore::InternalWritableStreamWriter::onClosedPromiseResolution):
(WebCore::InternalWritableStreamWriter::whenReady):
* Source/WebCore/bindings/js/InternalWritableStreamWriter.h: Added.

Canonical link: https://commits.webkit.org/300679@main
